cmd/resolve-version/main.go

```diff
@@ -55,6 +79,56 @@ func parseObject(key string) (release, error) {
 	return release{}, errors.New("Failed to parse key")
 }
 
+// Wrapper around the S3 API for listing objects
+// This maps directly to the API and parses the XML response but will not handle
+// paging and offsets automaticaly
+func fetchS3Result(bucketName string, options map[string]string) (result, error) {
+	var result result
+	v := url.Values{}
+	v.Set("list-type", "2")
+	for key, val := range options {
+		v.Set(key, val)
+	}
+	url := fmt.Sprintf("https://%s.s3.amazonaws.com?%s", bucketName, v.Encode())
+	resp, err := http.Get(url)
+	if err != nil {
+		return result, err
+	}
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return result, err
+	}
+
+	xml.Unmarshal(body, &result)
```

Comment:
should be `return result, xml.Unmarshal(body, &result)` i think. because `Unmarshal` can return err?

Reply:
Oh, that's a nifty idiom